fastapi_utils/base_route.py

```diff
@@ -0,0 +1,109 @@
+from typing import Dict, Generic, List, TypeVar
+
+from fastapi import Depends, HTTPException
+from pydantic import BaseModel
+from sqlalchemy.orm import Session
+
+from fastapi_utils.crud_base import Base, CRUDBase
+
+ResponseModelType = TypeVar("ResponseModelType", bound=BaseModel)
+ResponseModelManyType = TypeVar("ResponseModelManyType", bound=BaseModel)
+CreateSchemaType = TypeVar("CreateSchemaType", bound=BaseModel)
+UpdateSchemaType = TypeVar("UpdateSchemaType", bound=BaseModel)
+IDType = TypeVar("IDType")
+
+
+def get_filter_fields(self) -> List[str]:
+    """This would need to get overridden for each BaseRoute where the filter fields are defined.
+
+    Returns:
+        List[str] -- List of fields to filter by
+    """
+    return []
+
+
+class BaseRoute(Generic[ResponseModelType, ResponseModelManyType, CreateSchemaType, UpdateSchemaType, IDType]):
+    """A base route that has the basic CRUD endpoints.
+
+    For read_many
+
+    """
+
+    filter_fields: List[str] = Depends(get_filter_fields)
```

Comment:
Is there a benefit to this being a dependency rather than just a class var? That is to say, why not just use

```
    ...
    filter_fields: ClassVar[List[str]] = []
```
and then you can override the list in the subclass if desired? I think this should still work fine, with just a bit less overhead too.

Reply:
Oh that's exactly what I was looking for. Didn't know how to stop the class variables showing up in the openapi spec